cuda/test/base/array.cpp

```diff
@@ -2,15 +2,11 @@
 //
 // SPDX-License-Identifier: BSD-3-Clause
 
-#include <ginkgo/core/base/array.hpp>
-
-
 #include <gtest/gtest.h>
 
-
+#include <ginkgo/core/base/array.hpp>
```

Comment:
This is not the main header. The conversation about main headers only makes sense for files which implement functions defined in a header file. This is not the case for the test files, so there is no main header here.
I think we had a different interpretation of main header previously, which was based on *\~vibes\~*. Now we just follow clangs definition.
So all the comments on the test files are unnecessary. Sorry for not making that clearer before.

Reply:
with this definition, the opposite part needs to change.
`cuda/test/matrix/coo_kernels.cpp` uses `core/matrix/coo_kernels.hpp` as the main header because they are in the mainHeaderSource

Reply:
do you mean `cuda/matrix/coo_kernels.cpp`? Because there is no `cuda/test/matrix/coo_kernels.cpp`.

Reply:
sorry, I just randomly pick something might have issue.
something like this: `cuda/test/solver/lower_trs_kernels.cu` uses `core/solver/lower_trs_kernels.hpp`

Reply:
I see. This poses an issue, because there doesn't seem to be a way to prevent clang from detecting `core/solver/lower_trs_kernels.hpp` as main header. I think the simplest choice is to consider these headers as the main header again.

Side note: this particular header is unnecessary. The test doesn't call any kernel directly.

Reply:
Or we can just ignore it and live with these (small) inconsistencies. That means there will be some tests which have a main header and those that don't.